CMakeLists.txt

```diff
@@ -1,94 +1,242 @@
+    endif()
+
+    message(STATUS "  ${OPTION_NAME}: ${${OPTION_NAME}}")
+endmacro()
+
+macro(kompute_log_level OPTION_NAME OPTION_TEXT OPTION_DEFAULT)
+    set(${OPTION_NAME} ${OPTION_DEFAULT} CACHE STRING ${OPTION_TEXT})
+    set_property(CACHE ${OPTION_NAME} PROPERTY STRINGS "Trace" "Debug" "Info" "Warn" "Error" "Critical" "Default" "Off")
+
+    if(DEFINED ENV{${OPTION_NAME}})
+        # Allow setting the option through an environment variable
+        set(${OPTION_NAME} $ENV{${OPTION_NAME}})
+    endif()
+
+    if(${OPTION_NAME})
+        add_definitions(-D${OPTION_NAME})
+    endif()
+
+    # Allow disabling logging completely and prevent linking against it:
+    if(${KOMPUTE_OPT_LOG_LEVEL} STREQUAL "Off")
+        set(${OPTION_NAME}_DISABLED ON)
+        add_compile_definitions(${OPTION_NAME}_DISABLED=1)
+    endif()
+
+    message(STATUS "  ${OPTION_NAME}: ${${OPTION_NAME}}")
+endmacro()
+
+macro(kompute_option_string OPTION_NAME OPTION_TEXT OPTION_DEFAULT)
+    set(${OPTION_NAME} ${OPTION_DEFAULT} CACHE STRING ${OPTION_TEXT})
+
+    if(DEFINED ENV{${OPTION_NAME}})
+        # Allow setting the option through an environment variable
+        set(${OPTION_NAME} $ENV{${OPTION_NAME}})
+    endif()
+
+    if(${OPTION_NAME})
+        add_definitions(-D${OPTION_NAME})
+    endif()
+
+    message(STATUS "  ${OPTION_NAME}: ${${OPTION_NAME}}")
+endmacro()
+
+message(STATUS "General purpose GPU compute framework built on Vulkan")
+message(STATUS "=======================================================")
 
 # Enable or disable targets
-option(KOMPUTE_OPT_BUILD_TESTS "Enable if you want to build tests" 0)
-option(KOMPUTE_OPT_CODE_COVERAGE "Enable if you want code coverage" 0)
-option(KOMPUTE_OPT_BUILD_DOCS "Enable if you want to build documentation" 0)
-option(KOMPUTE_OPT_BUILD_SHADERS "Enable if you want to re-build all shader files" 0)
-option(KOMPUTE_OPT_BUILD_SINGLE_HEADER "Enable if you want to build the single header file" 0)
-option(KOMPUTE_OPT_INSTALL "Enable if you want to enable installation" 0)
+kompute_option(KOMPUTE_OPT_BUILD_TESTS "Enable if you want to build tests." OFF)
+kompute_option(KOMPUTE_OPT_CODE_COVERAGE "Enable if you want code coverage." OFF)
+kompute_option(KOMPUTE_OPT_BUILD_DOCS "Enable if you want to build documentation." OFF)
+kompute_option(KOMPUTE_OPT_INSTALL "Enable if you want to enable installation." OFF)
+
 # Build options
-option(KOMPUTE_OPT_BUILD_PYTHON "Enable if you want to build python bindings" 0)
-option(KOMPUTE_OPT_ENABLE_SPDLOG "Enable to compile with spdlog as the internal logging framework" 0)
-option(KOMPUTE_OPT_REPO_SUBMODULE_BUILD "Use the submodule repos instead of external package manager" 0)
-option(KOMPUTE_OPT_ANDROID_BUILD "Enable android compilation flags required" 0)
-option(KOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS "Explicitly disable debug layers even on debug" 0)
-option(KOMPUTE_OPT_DEPENDENCIES_SHARED_LIBS "Whether to use shared libraries for dependencies for install" 0)
```

Comment:
In my eyes everything should be build shared if possible. This has a couple of advantages:
* Less RAM usage in case multiple applications use the same lib. It has to be loaded into RAM only once.
* Makes patching easier without having to recompile completely in case we only want to change the minor version of the lib we are consuming.
* Most Packaging formats like .deb even require/enforce shared libs.

While on the other side static libs produce a large executable, that might be faster in some rare cases, although it is a single file.

In our case I think we should let the user decide how he wants to compile and don't mix and match static and shared libs.

Reply:
Ok those are some really good points, thanks for sharing. I would agree based on that. On th epoint about allowing user to decide how they want to compile (shared vs static) I completely agree that it should be either or as opposed to mix and match - only question from my side is, currently would there be a way to expose this as a build parmeter? (that was also my initial question as I saw that `KOMPUTE_OPT_DEPENDENCIES_SHARED_LIBS` was removed)

Reply:
Possible, but I don't like such an option since one should not mix and match build types. But in case you want, I can add such an option again :)